src/tmate.py

```diff
@@ -56,25 +63,61 @@ class FingerprintError(Exception):
     """Represents an error with generating fingerprints from public keys."""
 
 
-def install_dependencies() -> None:
-    """Install dependenciese required to start tmate-ssh-server container.
+def _setup_docker(proxy_config: typing.Optional[state.ProxyConfig] = None) -> None:
+    """Install and configure proxy settings for docker if available.
+
+    Args:
+        proxy_config: The proxy configuration to enable for dockerd.
 
     Raises:
-        DependencySetupError: if there was something wrong installing the apt package
-            dependencies.
+        DependencySetupError: if there was a problem installing/setting up Docker.
     """
+    if proxy_config:
+        environment = jinja2.Environment(
+            loader=jinja2.FileSystemLoader("templates"), autoescape=True
+        )
+        docker_template = environment.get_template("docker_daemon.json.j2")
+        daemon_config = docker_template.render(
+            HTTP_PROXY=proxy_config.http_proxy,
+            HTTPS_PROXY=proxy_config.https_proxy,
+            NO_PROXY=proxy_config.no_proxy,
+        )
+        DOCKER_DAEMON_CONFIG_PATH.parent.mkdir(parents=True, exist_ok=True)
+        DOCKER_DAEMON_CONFIG_PATH.touch(exist_ok=True)
+        DOCKER_DAEMON_CONFIG_PATH.write_text(daemon_config, encoding="utf-8")
+
     try:
-        apt.update()
-        apt.add_package(APT_DEPENDENCIES)
-    except (apt.PackageNotFoundError, apt.PackageError) as exc:
-        raise DependencySetupError("Failed to install apt packages.") from exc
+        apt.add_package("docker.io", update_cache=True)
+    except (apt.PackageNotFoundError, apt.PackageError, subprocess.CalledProcessError) as exc:
+        raise DependencySetupError("Failed to apt install Docker.") from exc
     passwd.add_group("docker")
     try:
         passwd.add_user_to_group(USER, "docker")
     except ValueError as exc:
```

Comment:
I'm not sure you really have to catch this. This looks more like a logical error than an actual exception. ValueError will be raised only if the user or the group do not exist

Reply:
Removed, thanks!